test/cookbooks/test/recipes/source_centos7.rb

```diff
@@ -0,0 +1,11 @@
+node.default['yum']['centos-sclo']['enabled'] = true
+node.default['yum']['centos-sclo']['managed'] = true
+node.default['yum']['centos-sclo-rh']['enabled'] = true
+node.default['yum']['centos-sclo-rh']['managed'] = true
+include_recipe 'yum-centos'
+
+yum_package 'devtoolset-7-gcc-c++'
+
+ENV['PATH'] = '/opt/rh/devtoolset-7/root/usr/bin:' + ENV['PATH']
+ENV['CC'] = '/opt/rh/devtoolset-7/root/usr/bin/gcc'
+ENV['CXX'] = '/opt/rh/devtoolset-7/root/usr/bin/g++'
```

Comment:
IMHO this should be implemented in the recipe for the cookbook and not in the test cookbook so that it can be used by users.

Reply:
Up to a point, I don't think this cookbook should mandate the use of a particular repo in the ordinary repositories.  I'm fine with setting up the repository in the test cookbook.

Reply:
thank you, @ramereth 
good point, it's easy to move this recipe to the cookbook.

Reply:
@ramereth do you think it's applicable to move a dependency `yum-centos` from test cookbook metadata.rb into the main one?

Reply:
hi @MarkGibbons 
i did not refresh the page and haven't seen your response. What if we move only `yum_package 'devtoolset-7-gcc-c++'` setup to main cookbook and converge fails if no repository configured in user's cookbook?

Reply:
I'd expect the user to configure the repository they need.  They may not have internet access.  The fix for centos should also apply to redhat  EL and the repositories are likely to be different.  Sorry, I'll look tonight.  Work is calling.  Really appreciate you looking into this.

Reply:
@MarkGibbons as a potential user, I'd expect this would work "out of the box". We could easily make it optional via an attribute to install/use the repository so that folks can use it more easily in air gap environments.